runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -98,7 +98,12 @@ function TSHighlighter:get_hl_from_capture(capture)
     return vim.split(name, '.', true)[1]
   else
     -- Default to false to avoid recomputing
-    return a.nvim_get_hl_id_by_name(TSHighlighter.hl_map[name])
+    local hl = TSHighlighter.hl_map[name]
+    if hl then
```

Comment:
personally I would `return (hl and a.nvim_get_hl_id_by_name(hl)) or 0` for a "unwrap" check like this.

Reply:
that's what i thought at first, but i wasn't sure if `nvim_get_hl_id_by_name` would error out if hl was undefined.
Also we could output some kind of warning message to tell users the query name is not recognized ?

Reply:
badly read the comment, indeed your version is better. We'll forget about outputting information for now, unless you wish to do it.